src/mini-cast-bar/hooks/useMiniPlayerMediaStatusListenerHook.js

```diff
@@ -0,0 +1,128 @@
+import { useEffect, useState } from "react";
+import {
+  VizbeeManager,
+  VizbeePlayerState,
+} from "../../../";
+import { ButtonType } from "../constants";
+import { LOG_TAG } from "../constants";
+
+export const useMiniPlayerMediaStatusListenerHook = ({
+  isVisible,
+  setIsVisible,
+  playActionButtonType,
+  shouldDisablePlayPauseButtonForLive,
+  buttonImgPlay,
+  buttonImgPause,
+  buttonImgStop,
+}) => {
+  // Define state
+  const [mediaStatus, setMediaStatus] = useState({});
+  const [isButtonDisabled, setButtonDisabled] = useState(true);
+  const [buttonImg, setButtonImg] = useState(null);
+  const [buttonType, setButtonType] = useState(null);
+
+  // useEffect hook
+  useEffect(() => {
+    console.debug(LOG_TAG, "Add VZB_MEDIA_STATUS Listener");
+    const mediaStatusListener = VizbeeManager.addListener(
+      "VZB_MEDIA_STATUS",
+      videoPlayerStatusCallback
+    );
+
+    return () => {
+      console.debug(LOG_TAG, "Remove VZB_MEDIA_STATUS Listener");
+      VizbeeManager.removeAllListeners(mediaStatusListener);
+    };
+  }, []);
+
+  /**
+   * Update the state of the mini player controls
+   *
+   * @param {Object} newMediaStatus - Media status containing video info
+   * @returns {void}
+   */
+  const updatePlayerState = (newMediaStatus) => {
+    console.debug(LOG_TAG, `New Media Status = ${newMediaStatus}`);
+    if (newMediaStatus?.playerState === VizbeePlayerState.Playing) {
+      setButtonDisabled(false);
+
+      if (newMediaStatus?.isLive && playActionButtonType !== "playpause") {
+        setButtonType(ButtonType.STOP);
+        setButtonImg(buttonImgStop);
+      } else {
+        if (newMediaStatus?.isLive && shouldDisablePlayPauseButtonForLive) {
+          setButtonDisabled(true);
+        }
+
+        setButtonType(ButtonType.PAUSE);
+        setButtonImg(buttonImgPause);
+      }
+      return;
+    }
+
+    if (newMediaStatus?.playerState === VizbeePlayerState.Paused || newMediaStatus?.playerState === VizbeePlayerState.Loading || newMediaStatus?.playerState === VizbeePlayerState.Buffering) {
```

Comment:
NitPick: let's break it down to multi line

Reply:
Done